hash_test.go

```diff
@@ -0,0 +1,53 @@
+package bn256
+
+import (
+	"testing"
+
+	"bytes"
+)
+
+func TestKnownHashes(t *testing.T) {
+	for i, mh := range marshaledHashes {
+		g := HashG1([]byte{byte(i)}, nil)
+		if !bytes.Equal(mh[:], g.Marshal()) {
+			t.Fatal("hash doesn't match a known value")
+		}
+	}
+}
+
+var buf = make([]byte, 8192)
+
+func benchmarkSize(b *testing.B, size int) {
+	b.SetBytes(int64(size))
+	for i := 0; i < b.N; i++ {
+		HashG1(buf[:size], nil)
+	}
+}
+
+func BenchmarkHashG1Size8bytes(b *testing.B) {
+	b.ResetTimer()
+	benchmarkSize(b, 8)
+}
+
+func BenchmarkHashG1Size1k(b *testing.B) {
+	b.ResetTimer()
+	benchmarkSize(b, 1024)
+}
+
+func BenchmarkHashG1Size8k(b *testing.B) {
+	b.ResetTimer()
+	benchmarkSize(b, 8192)
+}
+
+var marshaledHashes = [10][64]byte{
```

Comment:
```suggestion
var marshaledHashes = [11][64]byte{
```

Add another row for increasing coverage during tests.

Reply:
Added.